java/code/src/com/redhat/rhn/domain/cloudpayg/PaygProductFactory.java

```diff
@@ -82,6 +82,7 @@ public static void updateProducts(CloudCredentials credentials, Collection<PaygP
      * @param credentials the credentials
      * @return the list of products associated with the credentials or an empty list if none are set.
      */
+    @SuppressWarnings("java:S6204")
```

Comment:
Well, everybody using some linting and is not connected to our sonarcloud project will still see the "marked code" and use maybe the quickfix. This will result in errors as this function must return a modifiable list.
Adding a comment is ok, or we need to remove the streaming.

Reply:
added comments for now

Reply:
I'm not sure to understand though. You should not have any hint in this case because you can't replace with `toList()` as it's changing the behavior. Here some screenshot from IDE, this is the line we're talking about:

![image](https://github.com/user-attachments/assets/018bdb3c-aa70-4103-9973-5456bb183976)

This is for example another line you changed. Here since `toList()` is not changing the behavior I see the warning and the hint both from the IDE and sonarcloud:

![image](https://github.com/user-attachments/assets/bd0f573b-b8df-4190-9765-50c970c26a4d)

Reply:
I am not sure if this would be a better way if we write it

```java
    public static List<PaygProductInfo> getProductsForCredentials(CloudRMTCredentials credentials) {
        List<PaygProductInfo> credsId = getSession()
                .createNamedQuery("PaygCredentialsProduct.listByCredentialsId", PaygCredentialsProduct.class)
                .setParameter("credsId", credentials.getId())
                .stream()
                .map(p -> new PaygProductInfo(p.getName(), p.getVersion(), p.getArch()))
                .toList();
        return new ArrayList<>(credsId);
    }
```

Reply:
In fact you don't have to change, it's would be just wrong to do it.  This is a very bad suggestion IMHO, not sure why you see it and I don't, though. Maybe different plugin version? I'd honestly just rather disable this suggestion in SonarCloud, if it is applying to all occurrences of `collect(Collectors.toList())` without any check.

I run the intellj inspection and it changes only where the list is for sure not modified.

@rjmateus @rjpmestre Not sure what you think about this

Reply:
> I am not sure if this would be a better way if we write it
> 
> ```java
>     public static List<PaygProductInfo> getProductsForCredentials(CloudRMTCredentials credentials) {
>         List<PaygProductInfo> credsId = getSession()
>                 .createNamedQuery("PaygCredentialsProduct.listByCredentialsId", PaygCredentialsProduct.class)
>                 .setParameter("credsId", credentials.getId())
>                 .stream()
>                 .map(p -> new PaygProductInfo(p.getName(), p.getVersion(), p.getArch()))
>                 .toList();
>         return new ArrayList<>(credsId);
>     }
> ```

No, IMHO this code should not be changed at all. It's already ok. The suggestion is just wrong. There is no reason to prefer using `.toList()`. The `Collectors` method is not deprecated or anything, the new one is just a shorthand that does not make sense to apply here as you need a modifiable collection.

Reply:
I have disabled this rule in sonarcloud. It is not shown anymore in my IDE

Reply:
Look fine to me to just disable the rule in sonarcloud, since looks like we cannot have a consistent way of handling it.